lib/Serialization/SerializedModuleLoader.cpp

```diff
@@ -158,15 +160,10 @@ findModule(ASTContext &ctx, AccessPathElem moduleID,
                             moduleBuffer, moduleDocBuffer,
                             scratch);
 
-    if(err == std::errc::no_such_file_or_directory) {
-      addDiagnosticInfoForArchitectureMismatch(ctx,
-                                               moduleID.second,
-                                               moduleName,
-                                               archName,
-                                               currPath);
-    }
-
-    return false;
+      if(err == std::errc::no_such_file_or_directory) {
+        addDiagnosticInfoForArchitectureMismatch(
+            ctx, moduleID.second, moduleName, archName, currPath);
```

Comment:
Hm. I think we still *do* need to exit early, though (here and below). Otherwise we could go on to a different search path and find a different module with the same name.